src/main/java/pl/poznan/put/cs/scenariochecker/transformations/CountSpecialStepsScenarioStrategy.java

```diff
@@ -0,0 +1,25 @@
+package pl.poznan.put.cs.scenariochecker.transformations;
+
+import pl.poznan.put.cs.scenariochecker.model.Scenario;
+import pl.poznan.put.cs.scenariochecker.model.Step;
+
+import java.util.List;
+
+public class CountSpecialStepsScenarioStrategy extends ScenarioStrategy {
+
+    @Override
+    public void processScenario(Scenario scenario) {
+
+        scenario.setNumberOfSpecialSteps(countSpecialSteps(scenario.getSteps()));
+    }
+
+    private int countSpecialSteps(List<Step> steps) {
+        return  steps.
+                stream().
+                filter(ScenarioHelper::isSpecialStep).
+                map(step -> countSpecialSteps(step.getSubSteps()) + 1).
+                mapToInt(Integer::intValue).
+                sum();
```

Comment:
👍

Reply:
Really good job! It is the way that we wanted! Please fix formatting :stuck_out_tongue_winking_eye: